### PR TITLE
Added thumbnail fallback for obtaining icons for some pinned folders

### LINKED ISSUE #5423: Root folder without icon doesn't have an icon on sidebar (favorites)
**Explain the bug**
EDITION: It seems that they indeed have icons, but Files can't get it. This also happens with MTP root folders. So the bug is some root folders (smb, MTP) can't get their icon in Files.

**To reproduce**
1. Go to the root folder of a SMB / MTP folder. I'm using the root folder of a Synology NAS with SMB.
2. Right click and pin to favorites.
3. Root folder pinned to sidebar will not have an icon, but if you repeat those steps with subfolders, those will have it (check screenshot for better explanation).

**Expected behavior**
Should have an icon. Maybe a standard icon for any folder that doesn't have an icon?

**Logfile**
[debug.log](https://github.com/files-community/Files/files/6812040/debug.log)
[debug_fulltrust.log](https://github.com/files-community/Files/files/6812041/debug_fulltrust.log)

**Screenshots**
Windows explorer:
image

Files:
image

**System Information**
- OS Version: Windows 10 Pro 21H1 19043.1081 Windows Feature Experience Pack 120.2212.3530.0
- App version: 1.5.62.0

**Additional comment**
This problem still happen after restarting the app.

### PR DESCRIPTION
**Pull request by BanCrash**
**Resolved / Related Issues**
- Closes #5423

**Details of Changes**
- Added thumbnail fallback for fixing #5423 . Some folders doesn't have an icon on ListView but they have on SingleItem, so with this change more folders will have thumbnail icons when pinned. I added as a fallback because some folders have a bad thumbnail on SingleItem style (they have a black background).

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
Before:
image

After:
image

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@BanCrash can you split these changes into separate pull requests?

**Comment by BanCrash** [comment]
> @BanCrash can you split these changes into separate pull requests?

Sure, I did it for not creating a PR for just one line of code, but no problem!

**Comment by yaichenbaum** [comment]
@BanCrash understandable, it just makes it easier for us to reference changes in the future to have them separate 😅

**Comment by BanCrash** [comment]
> @BanCrash understandable, it just makes it easier for us to reference changes in the future to have them separate 😅

Ah, okay, I will do it that way in the future then :thumbsup:

**Review by gave92 — APPROVED** [review]

File: Files/DataModels/SidebarPinnedModel.cs [hunk 1]
```diff
@@ -269,6 +269,19 @@ public async Task AddItemToSidebarAsync(string path)
...
+                        Windows.Storage.FileProperties.ThumbnailOptions.ResizeThumbnail);
+
+                        if (thumbnailFallback != null)
+                        {
+                            locationItem.IconData = await thumbnailFallback.ToByteArrayAsync();
+                            locationItem.Icon = await locationItem.IconData.ToBitmapAsync();
+                        }
+                    }
                 }
                 else
```
Comment (gave92) [inline Files/DataModels/SidebarPinnedModel.cs:286]: @BanCrash could you check what happens if you revert the above change but change this line to `if (locationItem.IconData == null)`?
[Not mandatory, PR is ok as it is, I'm just curious]

**Comment by yaichenbaum** [comment]
Thank you!

**Comment by gave92** [comment]
For the record I meant this, no matter =)

```cs
if (res)
{
	using var thumbnail = await res.Result.GetThumbnailAsync(
		Windows.Storage.FileProperties.ThumbnailMode.ListView,
		24,
		Windows.Storage.FileProperties.ThumbnailOptions.ResizeThumbnail);
	if (thumbnail != null)
	{
		locationItem.IconData = await thumbnail.ToByteArrayAsync();
		locationItem.Icon = await locationItem.IconData.ToBitmapAsync();
	}
}
if (locationItem.IconData == null) <-- changed just this line, no need to fallback to SingleItem
{
	locationItem.IconData = await FileThumbnailHelper.LoadIconWithoutOverlayAsync(path, 24u);
	if (locationItem.IconData != null)
	{
		locationItem.Icon = await locationItem.IconData.ToBitmapAsync();
	}
}
```

**Comment by BanCrash** [comment]
> For the record I meant this, but no matter =)
> 
> ```cs
> if (res)
> {
> 	using var thumbnail = await res.Result.GetThumbnailAsync(
> 		Windows.Storage.FileProperties.ThumbnailMode.ListView,
> 		24,
> 		Windows.Storage.FileProperties.ThumbnailOptions.ResizeThumbnail);
> 	if (thumbnail != null)
> 	{
> 		locationItem.IconData = await thumbnail.ToByteArrayAsync();
> 		locationItem.Icon = await locationItem.IconData.ToBitmapAsync();
> 	}
> }
> if (locationItem.IconData == null) <-- changed just this line, no need to fallback to SingleItem
> {
> 	locationItem.IconData = await FileThumbnailHelper.LoadIconWithoutOverlayAsync(path, 24u);
> 	if (locationItem.IconData != null)
> 	{
> 		locationItem.Icon = await locationItem.IconData.ToBitmapAsync();
> 	}
> }
> ```

Ahh, okay, now I got it. I will try it and let you know ;)

**Comment by BanCrash** [comment]
@gave92 you were right:

image

**Comment by BanCrash** [comment]
@gave92 if you see my previous comment it was a mistake, I tested on the insider version XD

This is on the testing version. One is working but the other no:

image

**Comment by gave92** [comment]
@BanCrash that's interesting, thanks!

**Pull request merged**